tooling/lsp/src/requests/completion.rs

```diff
@@ -381,6 +1149,79 @@ impl<'a> NodeFinder<'a> {
         }
     }
 
+    fn builtin_functions_completion(&mut self, prefix: &str) {
+        if name_matches("assert", prefix) {
+            self.completion_items.push(snippet_completion_item(
+                "assert(…)",
+                CompletionItemKind::FUNCTION,
+                "assert(${1:predicate})",
+                Some("fn(T)".to_string()),
+            ));
+            self.completion_items.push(snippet_completion_item(
+                "assert(…)",
+                CompletionItemKind::FUNCTION,
+                "assert(${1:predicate}, ${2:message})",
+                Some("fn(T, str)".to_string()),
```

Comment:
Maybe we should change this to `fn<let N: u32>(T, str<N>)` ? Along with assert_eq below.

It did strike me as a bit odd when I saw the completion message and saw two listings for `assert`. What do other languages do for optional arguments?

Reply:
Ah, yes, I initially had only one overload (the one without a message). Then I realized two versions existed and I thought of suggesting both...

I just checked typescript and when it suggests a function name you just get the function name, not the parameters. Then when you type `(` you get hints for the parameters (we can do that, but it's a different LSP feature).

We could either show just one overload (the one without a message?) or not show these built-in functions... or something else? what's your preference?

Reply:
For now I kept only one overload (the one without a message), mainly for simplicity. We could revisit this minor issue later if needed.